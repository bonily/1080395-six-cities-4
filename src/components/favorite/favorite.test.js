import React from "react";
import renderer from "react-test-renderer";
import Favorite from "./favorite.jsx";

const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`
};

const offers = {
  Paris: [
    {
      id: 1,
      title: `Beautiful & luxurious apartment at great location`,
      description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.The building is green and from 18th century. An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
      price: 120,
      raiting: 4,
      bedrooms: 3,
      quests: {
        adults: 5,
        kids: 2
      },
      items: [`Wi-Fi`, `Heating`, `Kitchen`, `Cable TV`, `Baby seat`, `Dishwasher`, `Washing machine`],
      type: `apartment`,
      isInBookmark: false,
      isPremium: true,
      photos: [`img/1.png`, `img/2.png`],
      host: {
        avatar: `img/avatar-angelina.jpg`,
        name: `Angelina`,
        isSuper: true,
        avatarUrl: `img/avatar-max.jpg`,
      },
      coords: [52.3909553943508, 4.85309666406198],
      city: {
        name: `Paris`
      }
    },
    {
      id: 2,
      title: `Wood and stone place`,
      description: ``,
      price: 80,
      raiting: 4,
      bedrooms: 1,
      quests: {
        adults: 2,
        kids: 1
      },
      items: [`Wi Fi`, `Heating`, `Cable TV`, `Dishwasher`],
      type: `room`,
      isInBookmark: true,
      isPremium: false,
      photos: [`img/1.png`, `img/2.png`],
      host: {
        avatar: ``,
        name: `Masha`,
        isSuper: false,
        avatarUrl: `img/avatar-max.jpg`,

      },
      coords: [52.369553943508, 4.85309666406198],
      city: {
        name: `Amsterdam`
      }
    },
    {
      id: 3,
      title: `Canal View Prinsengracht`,
      description: ``,
      price: 132,
      raiting: 4,
      bedrooms: 6,
      quests: {
        adults: 10,
        kids: 5
      },
      items: [`Wi Fi`, `Heating`, `Kitchen`, `Cable TV`, `Fridge`],
      type: `house`,
      isInBookmark: false,
      isPremium: false,
      photos: [`img/1.png`, `img/2.png`],
      host: {
        avatar: ``,
        name: `Olga`,
        isSuper: true,
        avatarUrl: `img/avatar-olga.jpg`,
      },
      coords: [52.3909553943508, 4.929309666406198],
      city: {
        name: `Cologne`
      }
    },
    {
      id: 4,
      title: `Nice, cozy, warm big bed apartment`,
      description: ``,
      price: 180,
      raiting: 4,
      bedrooms: 1,
      quests: {
        adults: 2,
        kids: 0
      },
      items: [`Wi-Fi`, `Cable TV`, `Coffee machine`, `Washing machine`],
      type: `hotel`,
      isInBookmark: false,
      isPremium: false,
      photos: [`img/1.png`, `img/2.png`],
      host: {
        avatar: ``,
        name: `Roman`,
        isSuper: false,
        avatarUrl: `img/avatar-roma.jpg`,
      },
      coords: [52.3809553943508, 4.939309666406198],
      city: {
        name: `Barcelona`
      }
    }]
};

const NAME = `Petra`;


describe(`FavoriteSnapTest`, () => {
  it(`Favorite should render page with favorite offers`, () => {
    const tree = renderer
      .create(
          <Favorite
            favoriteOffers = {offers}
            onOfferTitleClick = {() => {}}
            authorizationStatus = {AuthorizationStatus.AUTH}
            onUserBlockClick = {() => {}}
            name = {NAME}
            сity = {`Paris`}
            changeFavoriteStatus = {() => {}}
            loadFavoriteOffers = {() => {}}
          />
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
