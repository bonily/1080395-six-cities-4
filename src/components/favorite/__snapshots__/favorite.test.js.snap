// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteSnapTest Favorite should render page with favorite offers 1`] = `
<div
  className="page"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="/img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/favorite"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__login"
                >
                  Petra
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--favorites"
  >
    <div
      className="page__favorites-container container"
    >
      <section
        className="favorites"
      >
        <h1
          className="favorites__title"
        >
          Saved listing
        </h1>
        <ul
          className="favorites__list"
        >
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <a
                  className="locations__item-link"
                  href="#"
                >
                  <span>
                    Paris
                  </span>
                </a>
              </div>
            </div>
            <div
              className="favorites__places"
            >
              <article
                className="favorites__place-card place-card"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                <div
                  className="place-card__mark"
                >
                  <span>
                    Premium
                  </span>
                </div>
                <div
                  className="favorites__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="#"
                  >
                    <img
                      alt="Place image"
                      className="place-card__image"
                      src="img/1.png"
                      style={
                        Object {
                          "height": "110px",
                          "width": "150px",
                        }
                      }
                    />
                     :
                  </a>
                </div>
                <div
                  className="place-card__info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        120
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button button"
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        To bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "80%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      className="place-card__name-link"
                      href="/offer/1"
                      onClick={[Function]}
                    >
                      Beautiful & luxurious apartment at great location
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    Apartment
                  </p>
                </div>
              </article>
            </div>
          </li>
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <a
                  className="locations__item-link"
                  href="#"
                >
                  <span>
                    Amsterdam
                  </span>
                </a>
              </div>
            </div>
            <div
              className="favorites__places"
            >
              <article
                className="favorites__place-card place-card"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                
                <div
                  className="favorites__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="#"
                  >
                    <img
                      alt="Place image"
                      className="place-card__image"
                      src="img/1.png"
                      style={
                        Object {
                          "height": "110px",
                          "width": "150px",
                        }
                      }
                    />
                     :
                  </a>
                </div>
                <div
                  className="place-card__info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        80
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button place-card__bookmark-button--active button"
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        To bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "80%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      className="place-card__name-link"
                      href="/offer/2"
                      onClick={[Function]}
                    >
                      Wood and stone place
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    Private room
                  </p>
                </div>
              </article>
            </div>
          </li>
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <a
                  className="locations__item-link"
                  href="#"
                >
                  <span>
                    Cologne
                  </span>
                </a>
              </div>
            </div>
            <div
              className="favorites__places"
            >
              <article
                className="favorites__place-card place-card"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                
                <div
                  className="favorites__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="#"
                  >
                    <img
                      alt="Place image"
                      className="place-card__image"
                      src="img/1.png"
                      style={
                        Object {
                          "height": "110px",
                          "width": "150px",
                        }
                      }
                    />
                     :
                  </a>
                </div>
                <div
                  className="place-card__info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        132
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button button"
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        To bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "80%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      className="place-card__name-link"
                      href="/offer/3"
                      onClick={[Function]}
                    >
                      Canal View Prinsengracht
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    House
                  </p>
                </div>
              </article>
            </div>
          </li>
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <a
                  className="locations__item-link"
                  href="#"
                >
                  <span>
                    Barcelona
                  </span>
                </a>
              </div>
            </div>
            <div
              className="favorites__places"
            >
              <article
                className="favorites__place-card place-card"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                
                <div
                  className="favorites__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="#"
                  >
                    <img
                      alt="Place image"
                      className="place-card__image"
                      src="img/1.png"
                      style={
                        Object {
                          "height": "110px",
                          "width": "150px",
                        }
                      }
                    />
                     :
                  </a>
                </div>
                <div
                  className="place-card__info"
                >
                  <div
                    className="place-card__price-wrapper"
                  >
                    <div
                      className="place-card__price"
                    >
                      <b
                        className="place-card__price-value"
                      >
                        €
                        180
                      </b>
                      <span
                        className="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      className="place-card__bookmark-button button"
                      onClick={[Function]}
                      type="button"
                    >
                      <svg
                        className="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlinkHref="#icon-bookmark"
                        />
                      </svg>
                      <span
                        className="visually-hidden"
                      >
                        To bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    className="place-card__rating rating"
                  >
                    <div
                      className="place-card__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "80%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    className="place-card__name"
                  >
                    <a
                      className="place-card__name-link"
                      href="/offer/4"
                      onClick={[Function]}
                    >
                      Nice, cozy, warm big bed apartment
                    </a>
                  </h2>
                  <p
                    className="place-card__type"
                  >
                    Hotel
                  </p>
                </div>
              </article>
            </div>
          </li>
        </ul>
      </section>
    </div>
  </main>
  <footer
    className="footer"
  >
    <a
      className="footer__logo-link"
      href="main.html"
    >
      <img
        alt="6 cities logo"
        className="footer__logo"
        height="33"
        src="img/logo.svg"
        width="64"
      />
    </a>
  </footer>
</div>
`;
